Core/include/Acts/Seeding/SeedFinder.ipp

```diff
@@ -538,19 +539,21 @@ void SeedFinder<external_spacepoint_t, platform_t>::filterCandidates(
       // the two seed segments using a scattering term scaled by the actual
       // measured pT (p2scatterSigma)
       float iHelixDiameter2 = B2 / S2;
-      // calculate scattering for p(T) calculated from seed curvature
-      float pT2scatterSigma = iHelixDiameter2 * options.sigmapT2perRadius;
-      // if pT > maxPtScattering, calculate allowed scattering angle using
-      // maxPtScattering instead of pt.
-      float pT = options.pTPerHelixRadius * std::sqrt(S2 / B2) / 2.;
-      if (pT > m_config.maxPtScattering) {
-        float pTscatterSigma = (m_config.highland / m_config.maxPtScattering) *
-                               m_config.sigmaScattering;
-        pT2scatterSigma = pTscatterSigma * pTscatterSigma;
-      }
       // convert p(T) to p scaling by sin^2(theta) AND scale by 1/sin^4(theta)
       // from rad to deltaCotTheta
-      float p2scatterSigma = pT2scatterSigma * iSinTheta2;
+      float p2scatterSigma = iHelixDiameter2 * sigmaSquaredSPtDependent;
+      if (!std::isinf(m_config.maxPtScattering)) {
+        // if pT > maxPtScattering, calculate allowed scattering angle using
+        // maxPtScattering instead of pt.
+        float pT = options.pTPerHelixRadius * std::sqrt(S2 / B2) / 2.;
+        if (pT > m_config.maxPtScattering) {
+          float pTscatterSigma =
+              (m_config.highland / m_config.maxPtScattering) *
+              m_config.sigmaScattering;
+          p2scatterSigma = pTscatterSigma * pTscatterSigma * iSinTheta2;
+        }
+      }
```

Comment:
it looks like the computation is actually different? In particular `sigmaSquaredSPtDependent` was not computed before and  `p2scatterSigma` at line 544 has an extra `iSinTheta2` then before?

If so, can you clarify (in the PR description is ok) what has changed?

Reply:
I believe the calculation is the same:
previously we were re-evaluating pT2scatterSigma within the if statement and then multipling it by iSinTheta2.
Now we also have only one iSinTheta2 factor because sigmaSquaredSPtDependent is not used when re-evaluating pT2scatterSigma within the if statement

so in both cases pT2scatterSigma = [(highland / maxPtScattering) * sigmaScattering ] ^2 * iSinTheta2

for ITk we don't get inside the if statement so this avoids some calculation